### PR TITLE
[smoke-tests] Fix vfn failover test to send transactions across public network to vfn instead of validator

### PR DESCRIPTION
**Pull request by bchocho**
### Description

* VFN was sending transactions to all validators using the public network.
* The test_vfn_failover smoke-test was nonsensical. It should actually fail to run the transactions if the VFN's validator is down. But managed to do so because the transactions were reaching other validators through the public network.

### Test Plan
Existing and modified smoke tests.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2248)

### CODE REVIEW

**Comment by bchocho** [comment]
Discussed offline with @zekun000. I've misinterpreted the vfn failover. I'll at least clean up this test to make more sense (adding VFNs instead of just leaving the validators with public networks)

File: testsuite/smoke-test/src/full_nodes.rs [hunk 1]
```diff
@@ -156,16 +157,18 @@ async fn test_vfn_failover() {
     }
 
     // Setup accounts
-    let mut account_0 = create_and_fund_account(&mut swarm, 100).await;
-    let account_1 = create_and_fund_account(&mut swarm, 100).await;
+    let mut account_0 = create_and_fund_account(&mut swarm, 10).await;
```
Comment (zekun000) [inline testsuite/smoke-test/src/full_nodes.rs:160]: why this change?

File: testsuite/smoke-test/src/full_nodes.rs [hunk 2]
```diff
@@ -179,16 +182,8 @@ async fn test_vfn_failover() {
...
-        transfer_coins_non_blocking(
-            &vfn_client,
-            &transaction_factory,
-            &mut account_0,
-            &account_1,
-            1,
-        )
-        .await;
-    }
+    assert_balance(&vfn_client, &account_0, 9).await;
```
Comment (zekun000) [inline testsuite/smoke-test/src/full_nodes.rs:185]: why is this assert instead of the transfer?

Comment (bchocho) [reply testsuite/smoke-test/src/full_nodes.rs:185]: These changes aren't too significant.

In this case, it seems more clear that we're checking that the transfer went through. (Even though transfer_coin also checks this, although not the balance).

**Review by zekun000 — APPROVED** [review]
